sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/common/exception/StatusRuntimeExceptionMapper.java

```diff
@@ -22,22 +22,80 @@
 import javax.ws.rs.core.Response;
 import org.jboss.resteasy.reactive.RestResponse;
+import org.jboss.resteasy.reactive.RestResponse.ResponseBuilder;
 import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
 
 /** Simple exception mapper for the {@link StatusRuntimeException}. */
 public class StatusRuntimeExceptionMapper {
 
   @ServerExceptionMapper
-  public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException exception) {
+  public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException grpcE) {
     // TODO finalize mapper, messages, logs, etc.
 
-    Status status = exception.getStatus();
-    ApiError body = new ApiError(status.getDescription());
+    Status.Code sc = grpcE.getStatus().getCode();
 
-    return switch (status.getCode()) {
-      case UNAVAILABLE -> RestResponse.status(Response.Status.BAD_GATEWAY, body);
-      case UNAUTHENTICATED -> RestResponse.status(Response.Status.UNAUTHORIZED, body);
-      default -> RestResponse.status(Response.Status.INTERNAL_SERVER_ERROR, body);
-    };
+    // Handling partly based on information from:
+    // https://developers.google.com/maps-booking/reference/grpc-api/status_codes
+    switch (sc) {
+        // Start with codes we know how to handle
+      case FAILED_PRECONDITION:
+        return grpcResponse(
+            sc,
+            Response.Status.BAD_REQUEST,
+            "Invalid state or argument(s) for gRPC operation",
+            grpcE.getMessage());
+      case INVALID_ARGUMENT:
+        return grpcResponse(
+            sc,
+            Response.Status.BAD_REQUEST,
+            "Invalid argument for gRPC operation",
+            grpcE.getMessage());
+      case NOT_FOUND:
+        return grpcResponse(
+            sc, Response.Status.NOT_FOUND, "Unknown gRPC operation", grpcE.getMessage());
+      case PERMISSION_DENIED:
+        return grpcResponse(
+            sc, Response.Status.UNAUTHORIZED, "Unauthorized gRPC operation", grpcE.getMessage());
+      case UNAUTHENTICATED:
+        return grpcResponse(
+            sc, Response.Status.UNAUTHORIZED, "Unauthenticated gRPC operation", grpcE.getMessage());
+      case UNAVAILABLE:
+        // Should this be SERVICE_UNAVAILABLE (503) or BAD_GATEWAY (502)?
+        return grpcResponse(
+            sc,
+            Response.Status.SERVICE_UNAVAILABLE,
+            "gRPC service unavailable",
+            grpcE.getMessage());
+      case UNIMPLEMENTED:
+        return grpcResponse(
+            sc,
+            Response.Status.NOT_IMPLEMENTED,
+            "Unimplemented gRPC operation",
+            grpcE.getMessage());
+
+        // And then codes we ... don't know how to handle
+      case OK: // huh?
+
+      case CANCELLED:
+      case UNKNOWN:
+      case DEADLINE_EXCEEDED:
+      case ALREADY_EXISTS:
+      case RESOURCE_EXHAUSTED:
+      case ABORTED:
+      case OUT_OF_RANGE:
+      case INTERNAL:
+      case DATA_LOSS:
+        break;
+    }
+
+    return grpcResponse(
```

Comment:
move to line 88 as default return for all, maybe a `default` case would make sense?